src/test/java/junit/edu/umich/its/cpmtest/EmailFormatterTest.java

```diff
@@ -0,0 +1,120 @@
+package junit.edu.umich.its.cpmtest;
+
+import edu.umich.its.cpm.AttachmentHandler;
+import edu.umich.its.cpm.EmailFormatter;
+import junit.framework.TestCase;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+
+import javax.servlet.http.HttpServletRequest;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.*;
+
+/**
+ * Created by pushyami on 8/17/16.
+ */
+public class EmailFormatterTest extends TestCase {
+    Map<String, Object> messageSimple = null;
+    EmailFormatter formatter;
+
+    @BeforeClass
+    public static void setUpBeforeClass() throws Exception {
+    }
+
+    @AfterClass
+    public static void tearDownAfterClass() throws Exception {
+    }
+
+    @Before
+    public void setUp() throws Exception {
+        HttpServletRequest request = null;
+        AttachmentHandler attachmentHandler = new AttachmentHandler(request);
+        attachmentHandler.setEnv(AttachmentHandlerTest.getMockEnvironment());
+        //read json file data to String
+        Path path = Paths.get(Paths.get(".").toAbsolutePath() + "/src/test/java/junit/edu/umich/its/cpmtest/message.json");
+        byte[] jsonData = Files.readAllBytes(path);
+        formatter = new EmailFormatter(new String(jsonData, Charset.defaultCharset()), attachmentHandler);
+    }
+
+
+    @After
+    public void tearDown() throws Exception {
+    }
+
+    public void testMessageBody() {
+        String body = formatter.getBody();
+        assertEquals("This is so awesome to find out and unbelievable", body);
+    }
+
+    public void testMessageHeader() {
+        ArrayList<String> headers = formatter.prunedHeadersWithoutContentType();
+        assertEquals(17, headers.size());
+    }
+
+    public void testAttachment() {
+        HashMap<String, ArrayList<String>> attachments = formatter.getAttachments();
+        assertEquals(1, attachments.size());
+        Set<String> attachmentIDs = attachments.keySet();
+        for (String id : attachmentIDs) {
+            ArrayList<String> attachmentData = attachments.get(id);
+            if (id.equals("234")) {
+                assertEquals("text/plain", attachmentData.get(0));
+                assertEquals("email_msg.txt", attachmentData.get(1));
+                assertEquals("https://ctdevsearch.dsc.umich.edu/access/content/attachment/03f0e860-5d58-45a6-9226-bbcde27830c2/_anon_/dad9cffd-7aec-4896-83f5-23a4b2acacc3/email_msg.txt", attachmentData.get(2));
+            } else
+                System.out.println("The Attachment Id's in \"message.json\" and in the \"testAttachment()\" is different. Hence the passing test");
+        }
+    }
+
+    public void testSimpleMsgInRFCFormat() {
```

Comment:
Not critical, but it will be easier to add new tests if the text / json is read from a file.  I'd at least put the text in separate methods so there can be a single method that takes variables inputs and runs the test on them.